### PR TITLE
ROX-7999: Fix SACv2 flake.

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Previously, we only triggered a blocking image scan, but missed to verify that the deployment has a risk calculated.

This led to flakes where the risk was not yet calculated at the point of asserting within the test.

Now, we ensure that the image is pre-scanned (which will cover a scan result being there and the risk upserting being faster) as well as that the deployment has a risk score calculated. 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
- CI (SACv2 tests should not flake anymore)

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 1]
```diff
@@ -62,8 +63,10 @@ class SACTest extends BaseSpecification {
     static final private Integer WAIT_FOR_VIOLATION_TIMEOUT = isRaceBuild() ? 600 : 60
 
     def setupSpec() {
-        // ROX-6260: pre scan the image to avoid missing risk score
-        Services.scanImage(TEST_IMAGE)
+        // ROX-6260: pre scan the image to avoid missing risk score.
```
Comment (rukletsov) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:66]: ROX-6260 is closed now and didn't give me any extra valuable info. I'd hence suggest to remove the reference; feel free to keep (or expand) the comment though.

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 2]
```diff
@@ -62,8 +63,10 @@ class SACTest extends BaseSpecification {
...
+        def img = Services.scanImage(TEST_IMAGE)
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:67]: Same test image is define in base specification maybe we can remove definition from this test

https://github.com/stackrox/stackrox/blob/7b98be2e0172026929d2e8e6908125e3c0bdba40/qa-tests-backend/src/test/groovy/BaseSpecification.groovy#L42

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 3]
```diff
@@ -73,6 +76,22 @@ class SACTest extends BaseSpecification {
...
+
+        // Make sure each deployment has a risk score.
+        def deployments = DeploymentService.listDeployments()
+        deployments.each {
+            def depID = it.id
+            try {
+                withRetry(30, 2) {
+                    assert DeploymentService.getDeploymentWithRisk(depID).hasRisk()
+                }
+            } catch (Exception e) {
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:88]: Do we need this catch? If getDeployment fails there is huge chance that test will fail but with hard to understand reason.

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 4]
```diff
@@ -73,6 +76,22 @@ class SACTest extends BaseSpecification {
...
+                if (strictIntegrationTesting) {
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:89]: What `strictIntegrationTesting` is for? If it's on we just rethrow exception but when it's fase we pass meanignful exception?

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 5]
```diff
@@ -73,6 +76,22 @@ class SACTest extends BaseSpecification {
...
+                    throw (e)
+                }
+                throw new AssumptionViolatedException("Failed to retrieve risk from deployment ${it.name}")
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:92]:
```suggestion
                throw new AssumptionViolatedException("Failed to retrieve risk from deployment ${it.name}", e)
```

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 6]
```diff
@@ -73,6 +76,22 @@ class SACTest extends BaseSpecification {
                 WAIT_FOR_VIOLATION_TIMEOUT)
         assert waitForViolation(DEPLOYMENT_QA2.name, "Secure Shell (ssh) Port Exposed",
                 WAIT_FOR_VIOLATION_TIMEOUT)
+
+        // Make sure each deployment has a risk score.
+        def deployments = DeploymentService.listDeployments()
+        deployments.each {
+            def depID = it.id
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:83]: You can name parameter 
```suggestion
        deployments.each { ListDeployment dep ->
```

File: qa-tests-backend/src/test/groovy/SACTest.groovy (hunk 4 above)
Comment (dhaus67) [reply qa-tests-backend/src/test/groovy/SACTest.groovy:89]: Hm I took this from `ImageScannigTests`, but I can remove it also if it doesn't really help at all.

**Review by janisz — APPROVED** [review]

**Pull request merged**